### PR TITLE
VBA Company Step (UI only)

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
Implements the UI portion of the VBA Company Step.

### Fixed Issues
Fixes (partial) https://github.com/Expensify/Expensify/issues/166916

### Tests / QA Steps
1. Go to `$expensifyCashURL/bank-account/`
1. Click on the bank icon
1. In the plaid modal, click "Chase"
1. Enter `user_good` and `pass_good` as login credentials.
1. Look at the shiny `Company Information` form on the next page.
1. Verify that each form element can be interacted with, and that the submit button or active form input is never covered by the software keyboard. 

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

#### Web
image

image

### CODE REVIEW

File: src/components/TextInputWithLabel.js [hunk 1]
```diff
@@ -1,25 +1,54 @@
...
     /** Text to show if there is an error */
     errorText: PropTypes.string,
+
+    /** Styles for the outermost container for this component. */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Text to use for a link in the label */
+    linkText: PropTypes.string,
+
+    /** URL to use for a link in the label */
```
Comment (marcaaron) [inline src/components/TextInputWithLabel.js:22]: >URL to use for a link shown after the label

File: src/components/TextInputWithLabel.js [hunk 2]
```diff
@@ -1,25 +1,54 @@
...
     /** Label text */
     label: PropTypes.string,
 
     /** Text to show if there is an error */
     errorText: PropTypes.string,
+
+    /** Styles for the outermost container for this component. */
+    containerStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Text to use for a link in the label */
```
Comment (marcaaron) [inline src/components/TextInputWithLabel.js:19]: > Text to use for a link shown after the label

File: src/components/TextInputWithLabel.js [hunk 3]
```diff
@@ -1,25 +1,54 @@
...
+    linkText: PropTypes.string,
```
Comment (marcaaron) [inline src/components/TextInputWithLabel.js:20]: NAB, I wonder if we should change this to `helpLinkText` so it's clear that the link shows up after the label itself

Another option would be to pass a `LabelComponent` like we are doing for the checkbox and not be so opinionated about what sorts of options a label will have... don't need to figure this out right now IMO.

**Review by marcaaron — COMMENTED** [review]
Looks pretty good to me! I like the light refactor of the drop down done here. Just had a small comment about the `linkText`. And will try some testing now.

File: src/pages/ReimbursementAccount/CompanyStep.js [hunk 4]
```diff
@@ -1,19 +1,108 @@
...
+import CheckboxWithLabel from '../../components/CheckboxWithLabel';
+import TextLink from '../../components/TextLink';
+import Picker from '../../components/Picker';
+import StatePicker from '../../components/StatePicker';
+import withLocalize, {withLocalizePropTypes} from '../../components/withLocalize';
 
-const CompanyStep = () => (
-    <View>
+const CompanyStep = ({translate}) => (
+    <ScrollView style={[styles.flex1, styles.w100]}>
```
Comment (marcaaron) [inline src/pages/ReimbursementAccount/CompanyStep.js:20]: Should the Header be inside the `ScrollView`? This way we can always access it? Probably the same would be true for our "Save & Continue" button at the bottom.

e.g. see the behavior of this compared to `/settings/profile`

https://user-images.githubusercontent.com/32969087/122284889-c3750680-ce89-11eb-949b-efa4398bc1d5.mp4

https://user-images.githubusercontent.com/32969087/122284899-c66ff700-ce89-11eb-974c-4fb5c7dea0bf.mp4

**Review by marcaaron — CHANGES_REQUESTED** [review]
Just a couple more comments after testing.  Everything looks great otherwise.

Navigating to `/settings/preferences` is giving me this error and white screen

Screen Shot 2021-06-16 at 9 59 07 AM

File: src/components/TextInputWithLabel.js (hunk 3 above)
Comment (roryabraham) [reply src/components/TextInputWithLabel.js:20]: Renamed to `helpLinkText`/`helpLinkURL`, but other than that we can refactor as needed later on.

Comment (NikkiWines) [reply src/components/TextInputWithLabel.js:20]: I like `helpLinkText`

**Review by NikkiWines — COMMENTED** [review]
This is looking good, definitely a lot I can reuse in the Requestor Step. A couple comments on the UI side: 

1. I think design prefers placeholders to labels for these forms (at least that was the direction we went in my PR after my first mockups had labels outside of the text fields)
Eg. 


2. I think we want to disable the `Save and Continue` button until all necessary fields have been filled out

File: src/pages/ReimbursementAccount/CompanyStep.js (hunk 4 above)
Comment (roryabraham) [reply src/pages/ReimbursementAccount/CompanyStep.js]: Good point, made this change 👍

**Comment by roryabraham** [comment]
Okay, @Expensify/design should I switch this form to placeholders instead of labels? We use labels elsewhere, such as in `settings/profile` and `settings/preferences`

**Comment by roryabraham** [comment]
> I think we want to disable the Save and Continue button until all necessary fields have been filled out

Good call-out, but right now there's no validation and the button doesn't do anything. Was going to handle this in a follow-up PR.

**Review by marcaaron — APPROVED** [review]
Looks good to me + can't reproduce the bug I ran into before. Thanks for the changes!

Would be great if we can create a follow up issue to track the date picker stuff.

**Review by NikkiWines — APPROVED** [review]
Don't want to hold this up so I'll approve, but I do think it would be good to have a solid direction re w/ or w/o labels for this flow. 

cc: @shawnborton @michelle-thompson @megankelso since you all commented on the design for another form in this VBA flow [here](https://github.com/Expensify/Expensify/issues/166919)

**Comment by roryabraham** [comment]
If we're going to discuss labels v.s: non-labels, I'd like to throw a suggestion in the ring (screenshot taken from my RN take-home challenge I did a few weeks ago.

image

Labels, but less clutter! best of both worlds

**Comment by shawnborton** [comment]
Heh, are you following along with the updated form style GH Rory? https://github.com/Expensify/Expensify.cash/issues/3108

**Comment by shawnborton** [comment]
Basically I think we have plans in the works to update the form styles across the board, and those plans involve keeping a label (in a much smoother fashion). So that being said, I'd say keep the labels for now and in the future we will convert them over to a much nicer component.

**Comment by NikkiWines** [comment]
Cool, given that I'll merge this and update my Requestor one to include the labels agian

**Pull request merged**